### ISSUE TITLE
readme.md clarification

### ISSUE THREAD

**Issue by stanwebber** [sycophantic/wyzeback#1]
can you please clarify this statement?

"The MAC must match the programmed MAC, it seems there firmware checks this against the MAC burned in CCFG"

are you saying we must use the mac originally printed on the detachable base or it won't work?  my sensors with 00000000 mac are hopelessly mixed up together at this point.  trial & error could run into the 100s of flashes.

**Comment by sycophantic** [comment]
Yes, it must be the MAC printed on the case or what you can read from the CCFG data region. You should be able to find the MAC by reading the flash.

**Comment by stanwebber** [comment]
thanks!  given the number of sensors involved this likely is my only viable option.  i have 0 experience with jtag type flashing so please don't shy away from being too rudimentary in your instructions.

i'm finding it increasing difficult to understand the mechanism for how the mac is lost in the first place.  the lost mac can be superimposed over a doner firmware so somehow it's lost from the original firmware, but is that really the case?  the firmware resides in flash and so does the CCFG which it performs a check against.  even with the lost mac the sensor is still able to function except that it can't be distinguished from any other sensor with a null or 00000000 mac.  my initial understanding was that the flash was somehow lost in a low/dead battery condition, but this would mean the entire firmware and CCFG would be lost which can't be the case.  i haven't examined your script, but you stress that your solution is a hack so i have to think you're modifying something else to circumvent the condition of the lost mac and that once the mac is lost from whatever critical location it's stored in--it's lost forever?

**Comment by sycophantic** [comment]
I compared a working vs. lost MAC firmware image and the places where the MAC was became null. I don't know why, I just know a find and replace worked only for the MAC that is in the CCFG.

**Comment by stanwebber** [comment]
absolutely bizarre.  has me wondering how the battery is even involved

**Comment by dannutu** [comment]
Kudos, really happy to see somebody finally managed to sort this out!

And another "readme clarification": WRT to the required CC1310 Launchpad, could you please confirm if the CC1310 LaunchPad "Evaluation Module" part no. LAUNCHXL-CC1310 would work or is one of the more expensive US/EU TI parts needed?
RF Development Tools LAUNCHXL-CC1310-1190 US  part no. LAUNCHXL-CC13-90US
RF Development Tools LAUNCHXL-CC1310-1190 EU  part no. LAUNCHXL-CC13-90EU
https://www.mouser.ie/Search/Refine?N=4294759686&Keyword=LAUNCHXL-CC1310

Cheers

**Comment by sycophantic** [comment]
@dannutu The cheap one works. That's exactly what I used, the launchpad eval module. You remove the jumpers and just connect it to the sensor.

**Comment by exeljb** [comment]
The python script didn't work for me on my mac (i tried using terminal) so instead, I manually changed the AABBCCDD mac address using find/replace in BBedit and it worked like a charm.  Just in case someone out there has that same issue

**Comment by dannutu** [comment]
> @dannutu The cheap one works. That's exactly what I used, the launchpad eval module. You remove the jumpers and just connect it to the sensor.

@sycophantic I had the CC1310 Launchpad since January and I finally took some time today to try to resurrect my dead Wyze contact sensors.
I installed Uniflash on a spare Windows machine and it detects the CC1310 Launchpad as CC1310F128. This is exactly what's printed on top of the small square chip in the middle of the Launchpad board so all is good so far. Once I select CC1310F128 Uniflash shows "Selected connection: Texas Instruments XDS110 Debug Bridge", which matches the ports under "Device Manager", which I think it's how it should be.
The first time I tried to "load" the patched firmware to the contact sensor Uniflash said it needed to update the firmware in the Launchpad itself. It failed the first time, I closed it and restarted it and then it worked and the board was successfully updated.
But I cannot get any further - no matter what I try in Uniflash (e.g. "more info", "erase flash" or "load") I get one of two errors:
`"Error connecting to the target: (Error -2131 @ 0x0) Unable to access device register"` or `"Error connecting to the target: (Error -242 @ 0x0) A router subpath could not be accessed"`
I googled, I even watched some youtube videos (from TI themselves and a few others) but I can't figure what's wrong and how can I make progress. The TI youtube video ("SimpleLink™ Wi-Fi ® CC3220 Getting Started with Project 0") was for another Launchpad board and an older version of Uniflash, it talked about having to put the Launchpad board into "Development mode" but I can't see that option in the latest version of Uniflash from the TI web site. Not sure at all if it's still needed or it's just old info nowadays. The Uniflash "help" opens a web page from the TI site, there is no mention of any errors codes, descriptions nor ways to fix them.
I followed the steps described here: https://forums.wyzecam.com/t/unbricking-wyze-contact-sensor-pcb-reset-pin/146856/102
I configured the jumpers as in this picture https://aws1.discourse-cdn.com/wyze/optimized/3X/9/e/9eaf0b965af1fb96817e1ac708d59d8d36c936bd_2_440x499.jpeg
When the Wyze contact sensor gets 3.3V via the Launchpad board the red LED blinks twice.
The magnet makes the Wyze contact sensor blink as expected. 
When trying to erase the flash or load the patched firmware the magnet is touching the sensor.
I even changed the USB cables between the Windows machine and the Launchpad board - nothing helped.

I've run out of ideas here, would you by any chance have a hint, please? Cheers!

**Comment by dannutu** [comment]
> The python script didn't work for me on my mac (i tried using terminal) so instead, I manually changed the AABBCCDD mac address using find/replace in BBedit and it worked like a charm. Just in case someone out there has that same issue

@exeljb I closely followed the steps you described on the wyzecam forum but I'm stuck getting `"Error connecting to the target: (Error -2131 @ 0x0) Unable to access device register"` or `"Error connecting to the target: (Error -242 @ 0x0) A router subpath could not be accessed"`, do you have any idea(s)? It's my very first contact with the Launchpad board and Uniflash and I have to say it's a bit daunting. Many thanks!

**Comment by dannutu** [comment]
BTW, I can see cJTAG comms over TCK and TMS using a protocol analyser whenever I try to "erase flash", "load" or "more info", however the protocol decoder throws an error. I don't yet know if that's because the sampling rate of 12 MHz is not fast enough for cJTAG, the computer is running out of memory before capturing all the cJTAG data, or maybe something else.

**Comment by dannutu** [comment]
Ha, I figured it out, it was the windows firewall, for whatever reason it needs **inbound** access. Once I allowed that I could reflash the 2 sensors and they are both working again now. Thanks for figuring out how to do this, is there a way I could buy you a virtual coffee or something like that?
